demos/multi_material/thermochemical_buoyancy/van_keken_1997.py

```diff
@@ -0,0 +1,313 @@
+# Rayleigh-Taylor instability
+# ===========================
+#
+# Rationale
+# ---------
+#
+# One may wish to simulate a geodynamical flow involving multiple physical phases. A
```

Comment:
For introduction to this tutorial, I think you can be less verbose, building on the previous example.